app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\User;
+use App\Role;
+use DB;
+use Session;
+use Hash;
+use Input;
+use App\Http\Requests\CreateUserRequest;
+use App\Http\Requests\UpdateUserRequest;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $users = User::orderBy('id', 'desc')->paginate(10);
+
+        return view('manage.users.index')->withUsers($users);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $roles = Role::all();
+
+        return view('manage.users.create')->withRoles($roles);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CreateUserRequest $request)
+    {
+        if (!empty($request->password)) {
+            $password = trim($request->password);
+        } else {
+            # set the manual password
+            $length = 10;
+            $keyspace = '123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ';
+            $str = '';
+            $max = mb_strlen($keyspace, '8bit') - 1;
+            for ($i = 0; $i < $length; ++$i) {
+
+                $str .= $keyspace[random_int(0, $max)];
+            }
+            $password = $str;
+        }
+
+        $user = new User();
+        $user->name = $request->name;
+        $user->email = $request->email;
+        $user->password = Hash::make($password);
+        $user->save();
+        if ($request->roles) {
+            $user->syncRoles($request->roles);
+        }
+
+        return redirect()->route('users.show', $user->id);
+
+        // if () {
+        //
+        // } else {
+        //   Session::flash('danger', 'Sorry a problem occurred while creating this user.');
+        //   return redirect()->route('users.create');
+        // }
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $user = User::where('id', $id)->with('roles')->first();
+
+        return view("manage.users.show")->withUser($user);
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $roles = Role::all();
+        $user = User::where('id', $id)->with('roles')->first();
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 106</br>String "manage.users.edit" does not require double quotes; use single quotes instead